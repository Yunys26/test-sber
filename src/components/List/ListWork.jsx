import React, { useEffect } from 'react';
// Libs
import {
    Typography,
    Grid,
    IconButton,
    Box, 
    CssBaseline, 
    ThemeProvider
} from '@material-ui/core';
import FavoriteIcon from '@material-ui/icons/Favorite';
import PropTypes from 'prop-types';
import moment from 'moment';
import { useDispatch } from 'react-redux';
// Theme
import { themeTabs } from '../TabsBlock/tabsBlockTheme';
import { showListLocalStorage } from '../../store/slicesStore/mainSlice';

import { addAndDelLocal, updateEffectLocalStorageStore } from '../../store/slicesStore/middleWareMainSlice';

export default function ListWork({ classes, data, statusList, local, showDataText }) {;

    const dispatch = useDispatch();

    // Обновление хранилища localStorageStore
    useEffect(() => {
        updateEffectLocalStorageStore(local, dispatch);
    },[]);
    
    
    return (
        <div>
            {
                (data.length === 0 && statusList) || data.map((el, index) =>
                    <Box className={classes.blockContent} key={el.dataDescription.id}>

                        <Box className={classes.workBlock} boxShadow={4}>
                        
                            <Grid 
                                container 
                                item={true} 
                                className={classes.titleSearchBlock} 
                                xs={12} 
                                direction="row" 
                                justify="space-between" 
                                alignItems="center"
                            >
                                <Typography className={classes.nameCompany} variant="h3">{el.dataDescription.company}</Typography>
                                {el.dataDescription.company_logo && <img className={classes.workBlockLogo} src={el.dataDescription.company_logo} alt="Logo Company"></img>}
                                <IconButton
                                    id={el.dataDescription.id}
                                    style={{color: local[el.id] && "#f50057"} || null}
                                    className={classes.likeBlack} 
                                    onClick={() => {
                                        dispatch(showListLocalStorage(addAndDelLocal(el.dataDescription.id, index, data, local)));
                                    }}>
                                    <FavoriteIcon className={classes.likeBlack} />
                                </IconButton>

                            </Grid>

                            { showDataText(el.dataDescription) }

                            <ThemeProvider theme={themeTabs}>
                            <CssBaseline/>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.dataDescription.description }}></div>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.dataDescription.how_to_apply }}></div>
                            </ThemeProvider>
                            <Typography className={classes.footerSearchBlock} variant="h6">{data.length !== 1 && moment(el.dataDescription.created_at).format('LLLL')}</Typography>
                        
                        </Box>
                        
                    </Box>
                )
            }
        </div> 
    )
}

ListWork.propTypes = {
    classes: PropTypes.object,
    data: PropTypes.array,
    statusList: PropTypes.object,
    local: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
    showDataText: PropTypes.func,
}