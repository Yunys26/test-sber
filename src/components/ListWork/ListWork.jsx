import { Box, CssBaseline, ThemeProvider } from '@material-ui/core';
import React from 'react';
// Libs
import {
    Typography,
    Grid,
    IconButton,
} from '@material-ui/core';
import FavoriteIcon from '@material-ui/icons/Favorite';
import moment from 'moment';
import { useDispatch } from 'react-redux';
// Theme
import { themeSearchBlock } from '../MainBlock/mainBlockTheme';
import { themeTabs } from '../TabsBlock/tabsBlockTheme';
import { openAndCloseModal} from '../../store/slicesStore/mainSlice';

export default function ListWork(props) {

    const { classes, data, funcLogic, statusList, statusListFavorit, colorButtonIcon } = props;

    const dispatch = useDispatch();

    return (
        <>
            {((data.length === 0 && statusList) || (data.length === 0 && statusListFavorit)) ||            
                data.map((el, index) =>
                    <Box key={index} className={classes.blockContent}>
                        <Box className={classes.workBlock} boxShadow={4} id={index} key={index}>
                            <Grid container item={true} className={classes.titleSearchBlock} xs={12} direction="row" justify="space-between" alignItems="center">
                                <Typography className={classes.nameCompany} variant="h3">{el.company}</Typography>
                                {el.company_logo !== null && <img className={classes.workBlockLogo} src={el.company_logo} alt=""></img> || null}
                                <ThemeProvider theme={themeSearchBlock}>
                                <CssBaseline/>
                                    <IconButton key={index} style={ colorButtonIcon && {color: "#f50057"}} className={classes.likeBlack} onClick={() => (localStorage.length <= 3 && funcLogic(index)) || (localStorage.length > 3 && dispatch(openAndCloseModal(true)))} >
                                        <FavoriteIcon className={classes.likeBlack} />
                                    </IconButton>
                                </ThemeProvider>
                            </Grid>
                            <Typography variant="h5"><b>Title:</b> {el.title}</Typography>
                            <Typography variant="h5"><b>Company:</b> {el.company}</Typography>
                            <Typography variant="h5"><b>Location:</b> {el.location}</Typography>
                            <Typography variant="h5"><b>Type:</b> {el.type}</Typography>
                            <Typography variant="h5"><b>Location:</b> {el.location}</Typography>
                            <ThemeProvider theme={themeTabs}>
                            <CssBaseline/>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.description }}></div>
                                <div className={classes.descriptionBlock} dangerouslySetInnerHTML={{ __html: el.how_to_apply }}></div>
                            </ThemeProvider>
                            <Typography className={classes.footerSearchBlock} variant="h6">{moment(el.created_at).format('LLLL')}</Typography>
                        </Box>
                    </Box>
                )
            }
        </>
    )
}
