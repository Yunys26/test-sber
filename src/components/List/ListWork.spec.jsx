import React from 'react';
import { shallow } from 'enzyme';
import ListWork from './ListWork';


// Мокаем useDispatch
jest.mock('react-redux', () => ({
    useDispatch: () => jest.fn()
}));

describe('<ListWork />', () => {
    let props, componentListWork;

    beforeEach(() => {

        props = {
            classes: {},
            data: [
                {
                    id: "65197190-0116-414a-9406-ae150edccb57",
                    dataDescription: {
                        company: "NS",
                        company_logo: "https://jobs.github.com/rails/active_storage/blobs/eyJfcmFpbHMiOnsibWVzc2FnZSI6IkJBaHBBdUdMIiwiZXhwIjpudWxsLCJwdXIiOiJibG9iX2lkIn19--9413d6855d50ab98787e3589f63491eacd6c7e0b/NS%20CB.jpg",
                        company_url: "http://www.ns.nl",
                        created_at: "Tue Oct 13 10:25:20 UTC 2020",
                        description: "senior iOS developer NS App - Utrecht Utrecht",
                        favorite: false,
                        how_to_apply: "Solliciteer direct via de",
                        id: "65197190-0116-414a-9406-ae150edccb57",
                        location: "Utrecht",
                        title: "senior iOS developer NS App ",
                        type: "Full Time",
                        url: "https://jobs.github.com/positions/65197190-0116-414a-9406-ae150edccb57",
                    },
                }
            ],
            statusList: {},
            local: [{}]
        };

        componentListWork = shallow(<ListWork {...props} />)
    });

    afterEach(() => {
        props, componentListWork = undefined;
    });

    it('Проверка на рендер <ListWork />', () => {
        expect(componentListWork.length).toBe(1);
    });
    
    it('Snapshot ListWork', () => {
        expect(componentListWork).toMatchSnapshot();
    });

})