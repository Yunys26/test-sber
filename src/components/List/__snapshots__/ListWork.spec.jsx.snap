// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListWork /> Snapshot ListWork 1`] = `undefined`;
